Scripts/Python/islmLakeLightMeter.py

```diff
@@ -67,26 +68,31 @@ def OnServerInitComplete(self):
         SDLvalue = 0
         if stringSDLVarName:
             ageSDL.setNotify(self.key, stringSDLVarName.value, 0.0)
+            ageSDL.setNotify(self.key, LightMeterVis.value, 0.0)
             SDLvalue = ageSDL[stringSDLVarName.value][0]
-            
-        for i in LakeLightMeter:
-            i.sceneobject.draw.disable()
-            i.sceneobject.physics.suppress(True)
-            
-        LakeLightMeter[SDLvalue].sceneobject.draw.enable()
-        LakeLightMeter[SDLvalue].sceneobject.physics.suppress(False)
-        
-        
+            MeterVis = ageSDL[LightMeterVis.value][0]
+
+        if MeterVis:
+            for i in LakeLightMeter:
+                i.sceneobject.draw.disable()
+                i.sceneobject.physics.suppress(True)
+
+            LakeLightMeter[SDLvalue].sceneobject.draw.enable()
+            LakeLightMeter[SDLvalue].sceneobject.physics.suppress(False)
+
+
     def OnSDLNotify(self, VARname, SDLname, PlayerID, tag):
-        if VARname != stringSDLVarName.value:
+        if (VARname != stringSDLVarName.value and VARname != LightMeterVis.value):
             return
-            
+
         ageSDL = PtGetAgeSDL()
         SDLvalue = ageSDL[stringSDLVarName.value][0]
-        
-        for i in LakeLightMeter:
-            i.sceneobject.draw.disable()
-            i.sceneobject.physics.suppress(True)
-            
-        LakeLightMeter[SDLvalue].sceneobject.draw.enable()
-        LakeLightMeter[SDLvalue].sceneobject.physics.suppress(False)
+        MeterVis = ageSDL[LightMeterVis.value][0]
+
+        if MeterVis:
+            for i in LakeLightMeter:
+                i.sceneobject.draw.disable()
+                i.sceneobject.physics.suppress(True)
+
+            LakeLightMeter[SDLvalue].sceneobject.draw.enable()
+            LakeLightMeter[SDLvalue].sceneobject.physics.suppress(False)
```

Comment:
What if the meter is turned off?

Reply:
Then we don't want the objects being turned on or off. They would be all off by the standard bool vis python method.

Reply:
That seems rather confusing to me - I think it would be better to handle all cases in this script to keep the logic grouped together.

Reply:
I could do that by moving the metervis check down past the for block. Easy enough change in python and ill need to remove the standard vis stuff in the prp. Ill get in this tomorrow.